documentation-website/Writerside/topics/Table-Definition.md

```diff
@@ -0,0 +1,192 @@
+field. For example, the `IntIdTable` has default integer `id` primary key. If you want to change column set, add columns, 
+or change primary key name to a custom one, you need to override this field of the appropriate table class.
+
+For example, if you want to define the `name` column as a primary key, use the following code. The "Cities_name" string 
+will be used in actual SQL query.
+```kotlin
+// SQL: CONSTRAINT Cities_name PRIMARY KEY ("NAME")
+override val primaryKey = PrimaryKey(name, name = "Cities_name")
+```
+
+### Foreign Key
+
+The `FOREIGN KEY` SQL constraint links two tables. Foreign key is a column from one table that refers to the primary key 
+or columns with a unique index from another table. To configure the foreign key, use `reference()` or `optReference()` 
+method. The second one let the foreign key accept the `null` value.
+
+`reference()` and `optReference()` methods have several parameters:
+
+* `name: String` is a name for foreign key column, which will be used in actual SQL queries.
+* `ref: Column<T>` is a target column from another parent table.
+* `onDelete: ReferenceOption? = null` is an action to the case when linked row from a parent table can be deleted.
+* `onUpdate: ReferenceOption? = null` is an action to the case when value in a referenced column can be changed.
+* `fkName: String? = null` is a foreign key constraint name.
+
+Enum class `ReferenceOption` has four values:
+
+* `RESTRICT` is an option restricts changes on referenced column, and the default option for MySQL dialect.
+* `NO_ACTION` is the same as RESTRICT, and the default option for Oracle and SQL Server dialects.
+* `CASCADE` is an option allows updating or deleting the referring rows.
+* `SET_NULL` is an option sets null to the referring column values.
+
+Consider the following `Citizens` table. This table has the `name` and `city` columns. Since the `Cities` table has 
+configured `name` primary key, the `Citizens` table can refer to it by its `city` column, which is a foreign key. To 
+configure such reference and make it nullable, use the `optReference()` method:
+```kotlin
+object Citizens : IntIdTable() {
+    val name = varchar("name", 50)
+    val city = optReference("city", Cities.name, onDelete=ReferenceOption.CASCADE)
+}
+```
+
+If any `Cities` row will be deleted, the appropriate `Citizens` row will be deleted too.
+
+### Check
+
+The `CHECK` SQL constraint checks that all values in a column match the condition. Exposed supports the `check()` method. 
+You apply this method to a column and pass the appropriate condition to it.
+
+For example, to check that the `name` column contains strings that begins with capital letter, use the following code:
+```kotlin
+// SQL: CONSTRAINT check_Cities_0 CHECK (REGEXP_LIKE("NAME", '^[A-Z].*', 'c')))
+val name = varchar("name", 50).check { it regexp "^[A-Z].*" }
+```
+
+Some DBMS could not support any condition. For more information, see its documentations.
```

Comment:
What does this sentence mean? Some databases do not support the CHECK constraint altogether?

Reply:
Based on the context, yes, I assume it was meant to be related to check constraints specifically. I've written something less vague in the recent changes.